ui/pages/Send.tsx

```diff
@@ -1,25 +1,305 @@
-import React, { ReactElement, useState } from "react"
+import { isAddress } from "@ethersproject/address"
+import { selectAccountAndTimestampedActivities } from "@tallyho/tally-background/redux-slices/accounts"
+import {
+  selectGasEstimates,
+  updateTransactionOptions,
+} from "@tallyho/tally-background/redux-slices/transaction-construction"
+import React, { ReactElement, useCallback, useEffect, useState } from "react"
+import { useDispatch, useSelector } from "react-redux"
 import { useLocation } from "react-router-dom"
 import CorePage from "../components/Core/CorePage"
 import SharedAssetInput from "../components/Shared/SharedAssetInput"
 import SharedButton from "../components/Shared/SharedButton"
-import SharedNetworkFeeGroup from "../components/Shared/SharedNetworkFeeGroup"
+import SharedInput from "../components/Shared/SharedInput"
+import SharedSlideUpMenu from "../components/Shared/SharedSlideUpMenu"
+import { useBackgroundSelector } from "../hooks"
 
 interface SendLocationState {
-  token: string
+  token: {
+    name: string
+  }
+}
+interface GasOption {
```

Comment:
This is something we've been sluggish to do but are finally zeroing in on: prefer using `type` over `interface`:

```suggestion
type GasOption = {
```